hack/update-deps-licenses.sh

```diff
@@ -182,48 +188,98 @@ echo "Populating vendor/ directory..."
 "$go" mod vendor
 
 # Put the K8S LICENSE on top
-(
-echo "================================================================================"
-echo "= cert-manager licensed under: ="
-echo
-cat "${LICENSE_ROOT}/LICENSE"
-echo
-echo "= LICENSE $(md5sum < "${LICENSE_ROOT}/LICENSE" | awk '{print $1}')"
-echo "================================================================================"
-) > ${TMP_LICENSE_FILE}
+if [ -f "${LICENSE_ROOT}/LICENSE" ]; then
+  (
+    echo "================================================================================"
+    echo "= cert-manager licensed under: ="
+    echo
+    cat "${LICENSE_ROOT}/LICENSE"
+    echo
+    echo "= LICENSE $(md5sum < "${LICENSE_ROOT}/LICENSE" | awk '{print $1}')"
+    echo "================================================================================"
+  ) > "${TMP_LICENSE_FILE}"
+fi
+
+# only_contains_submodules will check whether a package is empty of files
+# and its subdirectories are all modules, which indicates we do not need
+# a LICENSE for this module as it is an indirect transitive dependency.
+only_contains_submodules () {
+  package="$1"
+  if [[ -z "$(find "${DEPS_DIR}/${package}/" -mindepth 1 -maxdepth 1 -type f)" ]]; then
+    # If the package does not contain files, check whether all subdirectories
+    # only contain modules.
+    while read -d "" -r SUBDIR; do
```

Comment:
Shouldn't the delimiter be `\0` to handle the null delimited input from  ` find ... -print0` below.

Reply:
I'm actually not 100% certain, but this was taken from the existing file so I'm reluctant to change it without fully understanding it, and would expect us to need to make this change upstream too.

FWIW, I did test this and it does work successfully (as evidenced by the output from running this script against this branch without the Cobra v1.0.0 bump)

Reply:
```
-d delim
                     The first character of delim is used to terminate the input line, rather than newline.  If delim is the empty string, read will terminate a line when it reads a NUL character.
```

Reply:
So it sounds like this is correct? Given `-print0` terminates strings with the NUL character, and the empty delimiter indicates it should read until encountering NUL characters?

Reply:
Yep, that's my reading of it.

Reply:
Ah, thanks @james-w I was reading the help text for the builtin read which doesn't mention that :+1: 

```
$ help read
...
      -d delim	continue until the first character of DELIM is read, rather
    		than newline

```